internal/pkg/exec/docker.go

```diff
@@ -159,3 +172,51 @@ func imageName(uri, tag string) string {
 	}
 	return fmt.Sprintf("%s:%s", uri, tag)
 }
+
+// IsEcrCredentialHelperEnabled return true if ecr-login is enabled either globally or registry level
+func (c DockerCommand) IsEcrCredentialHelperEnabled(uri string) bool {
+	// Make sure the program is able to obtain the home directory
+	splits := strings.Split(uri, "/")
+	if c.homePath != "" && len(splits) > 0 {
+		// Look into the default locations
+		pathsToTry := []string{filepath.Join(".docker", "config.json"), ".dockercfg"}
+		for _, path := range pathsToTry {
+			content, err := ioutil.ReadFile(filepath.Join(c.homePath, path))
+			if err != nil {
+				// if we can't read the file keep going
+				continue
+			}
+
+			config, err := parseCredFromDockerConfig(content)
+			if err != nil {
+				continue
+			}
+
+			if config.CredsStore == CredStoreECRLogin || config.CredHelpers[splits[0]] == CredStoreECRLogin {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func parseCredFromDockerConfig(config []byte) (*dockerConfig, error) {
+	cred := dockerConfig{}
+	err := json.Unmarshal(config, &cred)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cred, nil
+}
+
+func userHomeDirectory() string {
+	home, err := os.UserHomeDir()
+	if err!= nil{
+		println("authenticate to ECR: %w", err)
```

Comment:
Can we remove this line? We don't log outside of the `cli` pkg. Or you might want to define a new error and catch that in the `cli` pkg.

Reply:
Removed